woocommerce/woocommerce-gutenberg-products-block
# Convert Product components to TS

Note this also includes some changes from #4091 - I plan to rebase when that has been merged.


This PR converts the following components to TypeScript:
- `Policies`
- `ProductBackorderBadge`
- `ProductBadge`
- `ProductDetails`
- `ProductLowStockBadge`
- `ProductImage`
- `ProductMetadata`
- It also adds a new type: `ProductResponseItemData` which defines how `itemData` on a cart item is shaped. (This is used by things like Product Add-Ons).

### How to test the changes in this Pull Request:

1. To begin make three products:
1. To make the Low Stock badge appear, edit a product and tick **Inventory > Enable stock management at product level** and make the low stock threshold lower than the current stock level.
2. To make a Product Backorder badge appear edit a product and check **Inventory > Enable stock management at product level** and make the stock level 0, In `Allow backorders` choose `Allow, but notify customer`
3. Add some Add-ons to a product using Product Add-ons.
4. Add each of these products to your cart and go to the cart block, ensure the cart works and all content appears as expected.
